dashel/dashel-win32.cpp

```diff
@@ -933,13 +945,69 @@ namespace Dashel
 			}
 
 			hf = CreateFileA(devName.c_str(), GENERIC_WRITE | GENERIC_READ, 0, NULL, OPEN_EXISTING, FILE_FLAG_OVERLAPPED, NULL);
-			if(hf == INVALID_HANDLE_VALUE)
-				throw DashelException(DashelException::ConnectionFailed, GetLastError(), "Cannot open serial port.");
+			if (hf == INVALID_HANDLE_VALUE)
+				throw DashelException(DashelException::ConnectionFailed, GetLastError(),
+					(std::string("Cannot open serial port ") + devName + ".").c_str());
 
 			buildDCB(hf, target.get<int>("baud"), target.get<int>("bits"), target.get<bool>("dtr"), target.get("parity"), target.get("stop"), target.get("fc"));
 
 			startStream(EvPotentialData);
 		}
+
+		virtual bool checkConnection()
```

Comment:
If this function does not affect the object (i.e. it is only a check), it should be marked `const`.